### PR TITLE
port to v2

### PR DESCRIPTION
**Pull request by Torinbob**
Hi,

You mentioned you were quite busy so I took the liberty of forking your repo and making a few minor tweaks to your code to get it ported to v2 (with the help of MarcBoule) - hope that's OK!

Teleport is an essential part of my Rack template and btn is super useful.

thanks,
Steve

### CODE REVIEW

File: src/Util.hpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // some utility functions
 
-#include "rack0.hpp" //TODO: ??? - stringf -> string::f
+#include "rack.hpp" //TODO: ??? - stringf -> string::f
```
Comment (mgunyho) [inline src/Util.hpp]: let's remove the `//TODO: ???` from here, we've changed `stringf` to `string::f` and there doesn't seem to be anything else to do when changing from rack0.hpp to rack.hpp.

File: src/Widgets.cpp [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #include "Widgets.hpp"
-#include "window.hpp" // WINDOW_MOD_CTRL
-#include <GLFW/glfw3.h> // key codes
+//#include "window.hpp" // WINDOW_MOD_CTRL
```
Comment (mgunyho) [inline src/Widgets.cpp]: Apparently `WINDOW_MOD_CTRL` and `GLFW/glfw3` are included somewhere else, do you know where? But anyway, we can remove these commented out lines entirely.

**Comment by mgunyho** [comment]
Hi,

I don't mind at all! In fact I'm very thankful that you went to the trouble of doing the port. Apparently there wasn't that much to do.

The code compiles fine for me with the v2 SDK, so it seems OK to merge. Let's keep it on a separate branch (`rack-v2`) until Rack v2 is officially released. I only have two small comments.

**Comment by mgunyho** [comment]
Ah, I realized that we also need to change the font loading, as discussed [here](https://vcvrack.com/manual/Migrate2#2-Potential-runtime-bugs). So that should still be changed in `Widgets.hpp/.cpp`.

File: src/Util.hpp (hunk 1 above)
Comment (Torinbob) [reply src/Util.hpp]: That's now done

File: src/Widgets.cpp (hunk 2 above)
Comment (Torinbob) [reply src/Widgets.cpp]: When someone asked about window.hpp on the forum the other day, Andrew replied:

"Including any Rack header file other than rack.hpp is unsupported/unstable. rack.hpp includes all of Rack’s plugin API"

So yes, safe to remove these entirely.

**Comment by Torinbob** [comment]
Font loading is now fixed for v2 so should hopefully should be good to go. Regarding the rack-v2 branch, I saw you made one in your repo, is there anything I need to do my end for that or can you just merge into your v2 from my Master?

thanks,
Steve

File: src/Widgets.cpp (hunk 2 above)
Comment (mgunyho) [reply src/Widgets.cpp]: Alright, thanks!

**Comment by mgunyho** [comment]
Thanks, looks good!

Yes, I made the branch and changed this PR to be merged to that branch, so no need to do anything else. After thinking about it a bit more, I'm not sure how Andrew wants to handle these branches when doing the migration. In the library repo migration issue there was talk about `v2` branches, maybe I'll have to rename it or something. But I think this can now be merged.

I'm also considering adding the v2 API features like port and light labels at some point, but that's for a later time.

**Pull request merged**